app.py

```diff
@@ -35,91 +36,105 @@
 if not os.path.exists(app.config['UPLOAD_FOLDER']):
     os.makedirs(app.config['UPLOAD_FOLDER'])
 
-def cleanup_old_files(base_filename):
-    """Removes any existing files with the same base name"""
-    file_paths = [
-        os.path.join(app.config['UPLOAD_FOLDER'], base_filename + '.webm'),
-        os.path.join(app.config['UPLOAD_FOLDER'], base_filename + '_isolated_audio.wav'),
-        os.path.join(app.config['UPLOAD_FOLDER'], base_filename + '_processed_video.mp4')
-    ]
-    for path in file_paths:
-        if os.path.exists(path):
-            logging.info(f"Removing old file: {path}")
-            os.remove(path)
-
-def sanitize_filename(filename):
-    # Replace non-alphanumerics apart from periods and underscores, with underscores
-    return re.sub(r'[^a-zA-Z0-9._-]', '_', filename)
-
-def download_youtube_video(url):
-    try:
-        # Extract media info first to sanitize title
-        with yt_dlp.YoutubeDL() as ydl:
-            info_dict = ydl.extract_info(url, download=False)
-            base_title = info_dict['title']
-            sanitized_title = sanitize_filename(base_title)
-
-        ydl_opts = {
-            'format': 'bestvideo+bestaudio/best',
-            'outtmpl': os.path.join(app.config['UPLOAD_FOLDER'], sanitized_title + '.%(ext)s'),
-            'noplaylist': True,
-            'keepvideo': True,  # Keep source files for merging
-            'n_threads': 6
-        }
-
-        with yt_dlp.YoutubeDL(ydl_opts) as ydl:
-            result = ydl.extract_info(url, download=True)
-
-            # Post-download file-ext gymnastics
-            if 'requested_formats' in result:
-
-                # If separate video and audio were downloaded and merged 
-                merged_ext = result['ext']
-            else:
-                # If a single file was downloaded; fallback to mp4 (which probably isn't the best idea)
-                merged_ext = result.get('ext', 'mp4')
 
-        # Return the sanitized file path with for processing
-        video_file = os.path.join(app.config['UPLOAD_FOLDER'], sanitized_title + "." + merged_ext)
-        return os.path.abspath(video_file)
+class Utils:
+    """Utility class for common operations like file cleanup and sanitization."""
+
+    @staticmethod
+    def cleanup_old_files(base_filename):
+        """Removes any existing files with the same base name."""
+        file_paths = [
```

Comment:
for `file_paths` utilize the `base_filename` to avoid repetition; something like
```
base_path = os.path.join(app.config['UPLOAD_FOLDER'], base_filename)
file_paths = [
    base_path + '.webm',
    # and same for the isolated audio and processed video
]
```